packages/cli/lib/cli.ts

```diff
@@ -164,7 +165,17 @@ export const generate = async (debug = false, inParentDirectory = false) => {
             const stripped = rendered.replace(/^\s+/, '');
 
             if (!fs.existsSync(`${dirPrefix}/${name}.ts`) && !fs.existsSync(`${dirPrefix}/${providerConfigKey}/${type}s/${name}.ts`)) {
-                fs.writeFileSync(`${dirPrefix}/${name}.ts`, stripped);
+                if (layout_mode === 'root') {
+                    fs.writeFileSync(`${dirPrefix}/${name}.ts`, stripped);
+                } else {
+                    if (!fs.existsSync(`${dirPrefix}/${providerConfigKey}`)) {
+                        fs.mkdirSync(`${dirPrefix}/${providerConfigKey}`);
+                    }
+                    if (!fs.existsSync(`${dirPrefix}/${providerConfigKey}/${type}s`)) {
+                        fs.mkdirSync(`${dirPrefix}/${providerConfigKey}/${type}s`);
+                    }
```

Comment:
I think `mkdirSync` has a `recursive` argument to create a full path without having to create each folder individually

Reply:
Thanks, updated